docs/extra/optimize_function.rst

```diff
@@ -0,0 +1,297 @@
+********************
+Optimizing functions
+********************
+
+The :py:mod:`sherpa.optmethods` module provides classes that let you
+optimize :doc:`models <../models/index>` when applyed to :doc:`data
+<../data/index>` instances, and this is handled by the :doc:`fit
+<../fit/index>` module.  You can apply the optimizers directly to
+functions if you do not need to take advantage of Sherpa's model and
+data classes.
+
+The optimizers in :py:mod:`sherpa.optmethods.optfcts` are functions
+which follow the same interface::
+
+  optimizer(cb, start, lolims, hilims, **kwargs)
+
+where `cb` is a callable that given a set or parameters returns the
+current statistic, `start` is the starting positions of the
+parameters, `lolims` and `hilims` are the bounds of the parameters,
+and the remaining keyword arguments are specific to the optimizer.
+
+Examples
+========
+
+The following imports have been made::
+
+  >>> import numpy as np
+  >>> from matplotlib import pyplot as plt
+  >>> from matplotlib import colors
+  >>> from mpl_toolkits.mplot3d import Axes3D
+  >>> from astropy.table import Table
+  >>> from sherpa.optmethods import optfcts
+
+A simple function
+-----------------
+
+The function we want to optimize is a version of the
+`Rosenbrock function <https://en.wikipedia.org/wiki/Rosenbrock_function>`_,
+where `a=2` and `b=10`, so the best-fit location is at::
+
+  x = 2
+  y = 2 * 2
+
+.. note::
+   The Sherpa test suite contains a number of test cases based on
+   functions like this, taken from
+   `More, J. J., Garbow, B. S., and Hillstrom, K. E.. Testing unconstrained optimization software. United States: N. p., 1978. Web. doi:10.2172/6650344. <https://www.osti.gov/biblio/6650344>`_.
+
+The function is::
+
+  def rosenbrock(x, y):
+      a, b = 2, 10
+      return (a - x) * (a - x) + b * (y - x * x) * (y - x * x)
```

Comment:
```suggestion
      return (a - x)**2 + b * (y - x * x)**2
```
personal preference. I think it's easier to read because it's shorter, but either way is fine.

Reply:
How about

```suggestion
      return (a - x)**2  + b * (y - x**2)**2   # :-)
```